### PR TITLE
Whois amendments

### PR DESCRIPTION
**Pull request by mgorny-morecode**
## Proposed Changes
Change docker image from `komand/python-3-37-plugin:3` to `komand/python-3-37-slim-plugin:3` | Use input constants | Changed `Exception` to `PluginException` | Added "f" strings

### Description

Describe the proposed changes:

  -

### Testing

Any testing information goes here.

## PR Requirements

Developers, verify you have completed the following items by checking them off:

### Style

Review the [style guide](https://komand.github.io/python/style.html)

- [x] For dependencies, pin [OS package](https://komand.github.io/python/style.html#dockerfile) and [Python package](https://komand.github.io/python/style.html#requirements-txt) versions
- [x] For security, set least privileged account with ``USER nobody`` in the ``Dockerfile`` when possible
- [x] For size, use the [slim SDK images](https://komand.github.io/python/sdk.html#sdk-versions) when possible: ``komand/python-3-37-slim-plugin`` and ``komand/python-3-37-plugin``
- [x] For error handling, use of [PluginException](https://komand.github.io/python/error_handling.html#plugin-exceptions) and [ConnectionTestException](https://komand.github.io/python/error_handling.html#connection-exceptions)
- [x] For logging, use [self.logger](https://komand.github.io/python/sdk.html#logging)
- [x] For docs, use [changelog style](https://komand.github.io/python/style.html#changelog)
- [x] For docs, validate markdown with ``make validate`` which calls ``mdl`` to lint ``help.md``

### Functional Checklist
- [x] Work fully completed
- [x] Functional
  - [x] Any new actions/triggers include JSON [test files](https://komand.github.io/python/style.html#tests) in the `tests/` directory created with `icon-plugin run -c sample $action > tests/$action.json`
  - [x] Tests should all pass unless it's a negative test. Negative tests have a naming convention of `tests/$action_bad.json`
  - [x] Unsuccessful tests should fail by raising an exception causing the plugin to die and an object should be returned on successful test
  - [x] Add functioning test results to PR, sanitize any output if necessary
    * Single action/trigger `icon-plugin run -T tests/example.json --debug --jq`
    * All actions/triggers shortcut `icon-plugin run -T all --debug --jq` (use PR format at end)
  - [x] Add functioning run results to PR, sanitize any output if necessary
    * Single action/trigger `icon-plugin run -R tests/example.json --debug --jq`
    * All actions/triggers shortcut `icon-plugin run -R all --debug --jq` (use PR format at end)

### Run

<details>

```
{
  "body": {
    "log": "rapid7/WHOIS:1.0.8. Step name: address\nInfo: Using registry: arin\n",
    "meta": {},
    "output": {
      "address": "1600 Amphitheatre Parkway",
      "cidr": "8.8.8.0/24",
      "city": "Mountain View",
      "country": "US",
      "netname": "LVLT-GOGL-8-8-8",
      "netrange": "8.8.8.0 - 8.8.8.255",
      "nettype": "Reallocated",
      "org_abuse_email": "network-abuse@google.com",
      "org_abuse_phone": "+1-650-253-0000",
      "org_tech_email": "arin-contact@google.com",
      "org_tech_phone": "+1-650-253-0000",
      "orgname": "Google LLC",
      "postal": "94043",
      "regdate": "2000-03-30",
      "state": "CA",
      "update": "2019-10-31"
    },
    "status": "ok"
  },
  "type": "action_event",
  "version": "v1"
}

```

<summary>
docker run --rm -i rapid7/whois:1.0.8 --debug run < tests/address.json
</summary>
</details>

<details>

```
{
  "body": {
    "log": "rapid7/WHOIS:1.0.8. Step name: address\nInfo: Using registry: apnic\n",
    "meta": {},
    "output": {
      "address": "7F, No.99, Huaihai Rd (E), Shanghai,200021",
      "cidr": "116.236.215.224 - 116.236.215.227",
      "country": "CN",
      "netname": "KAIYU",
      "netrange": "116.236.215.224 - 116.236.215.227",
      "nettype": "ASSIGNED NON-PORTABLE",
      "org_abuse_email": "huangyong@163.com",
      "org_abuse_phone": "+86-21-53063160-607",
      "orgname": "Shanghai Huangpu Kaiyu Culture Business Train Center",
      "update": "2009-11-09T08"
    },
    "status": "ok"
  },
  "type": "action_event",
  "version": "v1"
}

```

<summary>
docker run --rm -i rapid7/whois:1.0.8 --debug run < tests/address_apnic.json
</summary>
</details>

<details>

```
{
  "body": {
    "log": "rapid7/WHOIS:1.0.8. Step name: address\nInfo: Using registry: arin\n",
    "meta": {},
    "output": {
      "address": "1600 Amphitheatre Parkway",
      "cidr": "8.8.8.0/24",
      "city": "Mountain View",
      "country": "US",
      "netname": "LVLT-GOGL-8-8-8",
      "netrange": "8.8.8.0 - 8.8.8.255",
      "nettype": "Reallocated",
      "org_abuse_email": "network-abuse@google.com",
      "org_abuse_phone": "+1-650-253-0000",
      "org_tech_email": "arin-contact@google.com",
      "org_tech_phone": "+1-650-253-0000",
      "orgname": "Google LLC",
      "postal": "94043",
      "regdate": "2000-03-30",
      "state": "CA",
      "update": "2019-10-31"
    },
    "status": "ok"
  },
  "type": "action_event",
  "version": "v1"
}

```

<summary>
docker run --rm -i rapid7/whois:1.0.8 --debug run < tests/address_arin.json
</summary>
</details>

<details>

```
{
  "body": {
    "log": "rapid7/WHOIS:1.0.8. Step name: address\nInfo: Using registry: lacnic\n",
    "meta": {},
    "output": {
      "address": "2A - Panama -",
      "cidr": "186.74/15",
      "country": "PA",
      "netname": "PA-CWPA-LACNIC",
      "netrange": "186.74/15",
      "nettype": "allocated",
      "org_abuse_email": "networks@CWPANAMA.NET",
      "org_abuse_phone": "+507  2642238 []",
      "orgname": "Cable \u0026 Wireless Panama",
      "regdate": "20030428",
      "update": "20030430"
    },
    "status": "ok"
  },
  "type": "action_event",
  "version": "v1"
}

```

<summary>
docker run --rm -i rapid7/whois:1.0.8 --debug run < tests/address_lacnic.json
</summary>
</details>

<details>

```
{
  "body": {
    "log": "rapid7/WHOIS:1.0.8. Step name: address\nInfo: Using registry: ripe\n",
    "meta": {},
    "output": {
      "address": "1-st Frezernaya str. 2/1 korp. 2",
      "country": "RU",
      "netname": "VDSINA-NET",
      "netrange": "185.231.155.0 - 185.231.155.255",
      "nettype": "ASSIGNED PA",
      "org_abuse_email": "abuse@vdsina.ru",
      "orgname": "VDSINA VDS Hosting",
      "regdate": "2018-02-19T16",
      "updated": "2018-02-19T16"
    },
    "status": "ok"
  },
  "type": "action_event",
  "version": "v1"
}

```

<summary>
docker run --rm -i rapid7/whois:1.0.8 --debug run < tests/address_ripe.json
</summary>
</details>

<details>

```
{
  "body": {
    "log": "rapid7/WHOIS:1.0.8. Step name: domain\nGetting whois information for comedien.brussels\nlookup_results: \u003cwhois._3_adjust.Domain object at 0x7fc232a89be0\u003e\n",
    "meta": {},
    "output": {
      "creation_date": "2015-01-20T17:09:15",
      "dnssec": "unsigned",
      "domain_status": [
        "clienttransferprohibited https://icann.org/epp#clienttransferprohibited"
      ],
      "expiration_date": "2021-01-20T17:09:15",
      "last_updated": "2020-01-19T22:40:45",
      "name": "comedien.brussels",
      "name_servers": [
        "dns1.all2all.org",
        "dns4.all2all.org",
        "dns3.all2all.org",
        "dns2.all2all.org"
      ],
      "registrant_cc": "be",
      "registrar": "Gandi SAS",
      "registrar_abuse_contact_email": "abuse@support.gandi.net",
      "registrar_abuse_contact_phone": "+33.143731851",
      "registrar_iana_id": "81",
      "registrar_url": "https://www.gandi.net/domaine/gtld/brussels/news?lang=en",
      "registrar_whois_server": "registrar url: https://www.gandi.net/domaine/gtld/brussels/news?lang=en",
      "registry_domain_id": "d0000050214-brussels"
    },
    "status": "ok"
  },
  "type": "action_event",
  "version": "v1"
}

```

<summary>
docker run --rm -i rapid7/whois:1.0.8 --debug run < tests/brussels.json
</summary>
</details>

<details>

```
{
  "body": {
    "log": "rapid7/WHOIS:1.0.8. Step name: domain\nGetting whois information for fmovies.date\nlookup_results: \u003cwhois._3_adjust.Domain object at 0x7f548d378b38\u003e\n",
    "meta": {},
    "output": {
      "creation_date": "2019-10-25T02:15:04",
      "dnssec": "unsigned",
      "domain_status": [
        "clienttransferprohibited https://icann.org/epp#clienttransferprohibited"
      ],
      "last_updated": "2019-10-30T02:15:09",
      "name": "fmovies.date",
      "name_servers": [
        "derek.ns.cloudflare.com",
        "melinda.ns.cloudflare.com"
      ],
      "registrant_cc": "pa",
      "registrar": "NameCheap, Inc.",
      "registrar_abuse_contact_email": "abuse@namecheap.com",
      "registrar_abuse_contact_phone": "+1.6613102107",
      "registrar_iana_id": "1068",
      "registrar_url": "http://www.namecheap.com",
      "registrar_whois_server": "whois.namecheap.com",
      "registry_domain_id": "d49a39220c661405f8bf3bbd73990170a-nsr"
    },
    "status": "ok"
  },
  "type": "action_event",
  "version": "v1"
}

```

<summary>
docker run --rm -i rapid7/whois:1.0.8 --debug run < tests/date.json
</summary>
</details>

<details>

```
{
  "body": {
    "log": "rapid7/WHOIS:1.0.8. Step name: domain\nGetting whois information for google.com\nlookup_results: \u003cwhois._3_adjust.Domain object at 0x7f7adf138978\u003e\n",
    "meta": {},
    "output": {
      "creation_date": "1997-09-15T04:00:00",
      "dnssec": "unsigned",
      "domain_status": [
        "clientdeleteprohibited https://icann.org/epp#clientdeleteprohibited",
        "clienttransferprohibited https://icann.org/epp#clienttransferprohibited",
        "clientupdateprohibited https://icann.org/epp#clientupdateprohibited",
        "serverdeleteprohibited https://icann.org/epp#serverdeleteprohibited",
        "servertransferprohibited https://icann.org/epp#servertransferprohibited",
        "serverupdateprohibited https://icann.org/epp#serverupdateprohibited",
        "clientupdateprohibited (https://www.icann.org/epp#clientupdateprohibited)",
        "clienttransferprohibited (https://www.icann.org/epp#clienttransferprohibited)",
        "clientdeleteprohibited (https://www.icann.org/epp#clientdeleteprohibited)",
        "serverupdateprohibited (https://www.icann.org/epp#serverupdateprohibited)",
        "servertransferprohibited (https://www.icann.org/epp#servertransferprohibited)",
        "serverdeleteprohibited (https://www.icann.org/epp#serverdeleteprohibited)"
      ],
      "expiration_date": "2028-09-13T00:00:00-07:00",
      "last_updated": "2019-09-09T15:39:04",
      "name": "google.com",
      "name_servers": [
        "ns3.google.com",
        "ns2.google.com",
        "ns4.google.com",
        "ns1.google.com"
      ],
      "registrant_cc": "us",
      "registrar": "MarkMonitor Inc.",
      "registrar_abuse_contact_email": "abusecomplaints@markmonitor.com",
      "registrar_abuse_contact_phone": "+1.2083895740",
      "registrar_iana_id": "292",
      "registrar_url": "http://www.markmonitor.com",
      "registrar_whois_server": "whois.markmonitor.com",
      "registry_domain_id": "2138514_domain_com-vrsn"
    },
    "status": "ok"
  },
  "type": "action_event",
  "version": "v1"
}

```

<summary>
docker run --rm -i rapid7/whois:1.0.8 --debug run < tests/domain.json
</summary>
</details>

<details>

```
{
  "body": {
    "log": "rapid7/WHOIS:1.0.8. Step name: domain\nGetting whois information for 1app.online\nlookup_results: \u003cwhois._3_adjust.Domain object at 0x7f96396eaa58\u003e\n",
    "meta": {},
    "output": {
      "creation_date": "2019-10-01T19:47:25",
      "dnssec": "unsigned",
      "domain_status": [
        "clienttransferprohibited https://icann.org/epp#clienttransferprohibited"
      ],
      "expiration_date": "2020-10-01T23:59:59",
      "last_updated": "2019-11-25T10:16:27",
      "name": "1app.online",
      "name_servers": [
        "meera.ns.cloudflare.com",
        "burt.ns.cloudflare.com"
      ],
      "registrant_cc": "us",
      "registrar": "PDR Ltd. d/b/a PublicDomainRegistry.com",
      "registrar_abuse_contact_email": "abuse@publicdomainregistry.com",
      "registrar_abuse_contact_phone": "+1.2013775952",
      "registrar_iana_id": "303",
      "registrar_url": "https://publicdomainregistry.com",
      "registrar_whois_server": "whois.publicdomainregistry.com",
      "registry_domain_id": "d132574116-cnic"
    },
    "status": "ok"
  },
  "type": "action_event",
  "version": "v1"
}

```

<summary>
docker run --rm -i rapid7/whois:1.0.8 --debug run < tests/online.json
</summary>
</details>

<details>

```
{
  "body": {
    "log": "rapid7/WHOIS:1.0.8. Step name: domain\nGetting whois information for nic.zuerich\nlookup_results: \u003cwhois._3_adjust.Domain object at 0x7f10ed819b70\u003e\n",
    "meta": {},
    "output": {
      "creation_date": "2014-11-12T07:33:03",
      "dnssec": "signeddelegation",
      "domain_status": [
        "servertransferprohibited https://icann.org/epp#servertransferprohibited",
        "serverupdateprohibited https://icann.org/epp#serverupdateprohibited",
        "serverdeleteprohibited https://icann.org/epp#serverdeleteprohibited",
        "serverrenewprohibited https://icann.org/epp#serverrenewprohibited"
      ],
      "last_updated": "2019-05-27T08:27:40",
      "name": "nic.zuerich",
      "name_servers": [
        "anycast1.dnsres.net",
        "anycast2.dnsres.net"
      ],
      "registrant_cc": "ch",
      "registrar": "Kanton Z*rich",
      "registrar_abuse_contact_email": "abuse@dotzuerich.ch",
      "registrar_abuse_contact_phone": "+41.798319275",
      "registrar_iana_id": "9999",
      "registrar_url": "http://nic.zuerich",
      "registrar_whois_server": "whois.nic.zuerich",
      "registry_domain_id": "d96342284-cnic"
    },
    "status": "ok"
  },
  "type": "action_event",
  "version": "v1"
}

```

<summary>
docker run --rm -i rapid7/whois:1.0.8 --debug run < tests/zuerich.json
</summary>
</details>

### Test

Autogenerate with:
<details>

```
{
  "body": {
    "log": "rapid7/WHOIS:1.0.8. Step name: address\n",
    "meta": {},
    "output": null,
    "status": "ok"
  },
  "type": "action_event",
  "version": "v1"
}

```

<summary>
docker run --rm -i rapid7/whois:1.0.8 --debug test < tests/address.json
</summary>
</details>

Autogenerate with:
<details>

```
{
  "body": {
    "log": "rapid7/WHOIS:1.0.8. Step name: address\n",
    "meta": {},
    "output": null,
    "status": "ok"
  },
  "type": "action_event",
  "version": "v1"
}

```

<summary>
docker run --rm -i rapid7/whois:1.0.8 --debug test < tests/address_apnic.json
</summary>
</details>

Autogenerate with:
<details>

```
{
  "body": {
    "log": "rapid7/WHOIS:1.0.8. Step name: address\n",
    "meta": {},
    "output": null,
    "status": "ok"
  },
  "type": "action_event",
  "version": "v1"
}

```

<summary>
docker run --rm -i rapid7/whois:1.0.8 --debug test < tests/address_arin.json
</summary>
</details>

Autogenerate with:
<details>

```
{
  "body": {
    "log": "rapid7/WHOIS:1.0.8. Step name: address\n",
    "meta": {},
    "output": null,
    "status": "ok"
  },
  "type": "action_event",
  "version": "v1"
}

```

<summary>
docker run --rm -i rapid7/whois:1.0.8 --debug test < tests/address_lacnic.json
</summary>
</details>

Autogenerate with:
<details>

```
{
  "body": {
    "log": "rapid7/WHOIS:1.0.8. Step name: address\n",
    "meta": {},
    "output": null,
    "status": "ok"
  },
  "type": "action_event",
  "version": "v1"
}

```

<summary>
docker run --rm -i rapid7/whois:1.0.8 --debug test < tests/address_ripe.json
</summary>
</details>

Autogenerate with:
<details>

```
{
  "body": {
    "log": "rapid7/WHOIS:1.0.8. Step name: domain\n",
    "meta": {},
    "output": null,
    "status": "ok"
  },
  "type": "action_event",
  "version": "v1"
}

```

<summary>
docker run --rm -i rapid7/whois:1.0.8 --debug test < tests/brussels.json
</summary>
</details>

Autogenerate with:
<details>

```
{
  "body": {
    "log": "rapid7/WHOIS:1.0.8. Step name: domain\n",
    "meta": {},
    "output": null,
    "status": "ok"
  },
  "type": "action_event",
  "version": "v1"
}

```

<summary>
docker run --rm -i rapid7/whois:1.0.8 --debug test < tests/date.json
</summary>
</details>

Autogenerate with:
<details>

```
{
  "body": {
    "log": "rapid7/WHOIS:1.0.8. Step name: domain\n",
    "meta": {},
    "output": null,
    "status": "ok"
  },
  "type": "action_event",
  "version": "v1"
}

```

<summary>
docker run --rm -i rapid7/whois:1.0.8 --debug test < tests/domain.json
</summary>
</details>

Autogenerate with:
<details>

```
{
  "body": {
    "log": "rapid7/WHOIS:1.0.8. Step name: domain\n",
    "meta": {},
    "output": null,
    "status": "ok"
  },
  "type": "action_event",
  "version": "v1"
}

```

<summary>
docker run --rm -i rapid7/whois:1.0.8 --debug test < tests/online.json
</summary>
</details>

Autogenerate with:
<details>

```
{
  "body": {
    "log": "rapid7/WHOIS:1.0.8. Step name: domain\n",
    "meta": {},
    "output": null,
    "status": "ok"
  },
  "type": "action_event",
  "version": "v1"
}

```

<summary>
docker run --rm -i rapid7/whois:1.0.8 --debug test < tests/zuerich.json
</summary>
</details>

Autogenerate with:
<details>

```
[*] Validating plugin with all validators at .

[*] Running Integration Validators...
[*] Executing validator HelpValidator
[*] Executing validator ChangelogValidator
[*] Executing validator RequiredKeysValidator
[*] Executing validator UseCaseValidator
[*] Executing validator SpecPropertiesValidator
[*] Executing validator SpecVersionValidator
[*] Executing validator FilesValidator
[*] Executing validator TagValidator
[*] Executing validator DescriptionValidator
[*] Executing validator TitleValidator
[*] Executing validator VendorValidator
[*] Executing validator DefaultValueValidator
[*] Executing validator IconValidator
[*] Executing validator RequiredValidator
[*] Executing validator VersionValidator
[*] Executing validator DockerfileParentValidator
[*] Executing validator LoggingValidator
[*] Executing validator ProfanityValidator
[*] Executing validator AcronymValidator
[*] Executing validator JSONValidator
[*] Executing validator OutputValidator
[*] Executing validator RegenerationValidator
[*] Executing validator HelpInputOutputValidator
[*] Executing validator SupportValidator
[*] Executing validator ExceptionValidator
[*] Executing validator CredentialsValidator
[*] Executing validator PasswordValidator
[*] Executing validator PrintValidator
[*] Executing validator ConfidentialValidator
[*] Executing validator DockerValidator
[*] Executing validator URLValidator
[*] Plugin successfully validated!

----
[*] Total time elapsed: 64406.316000000006ms

```

<summary>
icon-validate --all .
</summary>
</details>

Autogenerate with:
<details>

```
[*] Use ``make menu`` for available targets
[*] Including available Makefiles: ../tools/Makefiles/Helpers.mk ../tools/Makefiles/Colors.mk
--
[*] Running validators
[*] Validating plugin at .

[*] Running Integration Validators...
[*] Executing validator HelpValidator
[*] Executing validator ChangelogValidator
[*] Executing validator RequiredKeysValidator
[*] Executing validator UseCaseValidator
[*] Executing validator SpecPropertiesValidator
[*] Executing validator SpecVersionValidator
[*] Executing validator FilesValidator
[*] Executing validator TagValidator
[*] Executing validator DescriptionValidator
[*] Executing validator TitleValidator
[*] Executing validator VendorValidator
[*] Executing validator DefaultValueValidator
[*] Executing validator IconValidator
[*] Executing validator RequiredValidator
[*] Executing validator VersionValidator
[*] Executing validator DockerfileParentValidator
[*] Executing validator LoggingValidator
[*] Executing validator ProfanityValidator
[*] Executing validator AcronymValidator
[*] Executing validator JSONValidator
[*] Executing validator OutputValidator
[*] Executing validator RegenerationValidator
[*] Executing validator HelpInputOutputValidator
[*] Executing validator SupportValidator
[*] Plugin successfully validated!

----
[*] Total time elapsed: 346.263ms

[*] Validating spec with js-yaml
[SUCCESS] Passes js-yaml spec check


[*] Validating markdown...
[SUCCESS] Passes markdown linting

[*] Validating python files for style...
[SUCCESS] Passes flake8 linting

[*] Validating python files for security vulnerabilities...
[main]  INFO    profile include tests: None
[main]  INFO    profile exclude tests: None
[main]  INFO    cli include tests: None
[main]  INFO    cli exclude tests: None
[main]  INFO    running on Python 3.7.6
Run started:2020-03-29 20:09:00.761559

Test results:
        No issues identified.

Code scanned:
        Total lines of code: 512
        Total lines skipped (#nosec): 0

Run metrics:
        Total issues (by severity):
                Undefined: 0.0
                Low: 0.0
                Medium: 0.0
                High: 0.0
        Total issues (by confidence):
                Undefined: 0.0
                Low: 0.0
                Medium: 0.0
                High: 0.0
Files skipped (0):
[SUCCESS] Passes bandit security checks

```

<summary>
make validate
</summary>
</details>

### CODE REVIEW

File: whois/help.md [hunk 1]
```diff
@@ -32,6 +32,16 @@ This action is used to retrieve data about a domain name.
 |----|----|-------|--------|-----------|----|
 |domain|string|None|True|Domain name to lookup|None|
 
+Example input:
+
+```
+
```
Comment (mrinehart-r7) [inline whois/help.md]:
```suggestion
```

File: whois/help.md [hunk 2]
```diff
@@ -32,6 +32,16 @@ This action is used to retrieve data about a domain name.
...
+{
+    "domain": "google.com"
+}
+
```
Comment (mrinehart-r7) [inline whois/help.md]:
```suggestion
```

File: whois/help.md [hunk 3]
```diff
@@ -85,6 +115,16 @@ This action is used to retrieve data about an IP address.
 |----|----|-------|--------|-----------|----|
 |address|string|None|True|IP to Lookup|None|
 
+Example input:
+
+```
+
```
Comment (mrinehart-r7) [inline whois/help.md]:
```suggestion
```

File: whois/help.md [hunk 4]
```diff
@@ -85,6 +115,16 @@ This action is used to retrieve data about an IP address.
...
+{
+  "address": "8.1.1.1"
+}
+
```
Comment (mrinehart-r7) [inline whois/help.md]:
```suggestion
```

File: whois/komand_whois/actions/domain/action.py [hunk 5]
```diff
@@ -1,23 +1,23 @@
-import komand
-from .schema import DomainInput, DomainOutput, Input
-
 # Custom imports below
 import whois
+
+import komand
 from komand.exceptions import PluginException
+from .schema import DomainInput, DomainOutput, Input, Component
```
Comment (mrinehart-r7) [inline whois/komand_whois/actions/domain/action.py]: Imports out of order

**Comment by jschipp-r7** [comment]
Sorry, @mgorny-morecode can you pull in latest code and fix the conflicts. We had some updates go out with example inputs.

**Pull request closed without merging**